rubicon/objc/objc.py

```diff
@@ -1696,3 +1726,62 @@ def __call__(self, *args):
 class ObjCBlockInstance(ObjCInstance):
     def __call__(self, *args):
         return self.block(*args)
+
+
+_NSConcreteGlobalBlock = ObjCClass("__NSGlobalBlock__")
+
+
+NOTHING = object()
+
+
+class Block:
+    def __init__(self, func, restype=NOTHING, *arg_types):
+        if not callable(func):
+            raise TypeError('Blocks must be callable')
+
+        self.func = func
+
+        argspec = inspect.getfullargspec(inspect.unwrap(func))
+
+        if restype is NOTHING:
+            try:
+                restype = argspec.annotations['return']
+            except KeyError:
+                raise ValueError(
+                    'Block callables must be fully annotated or an explicit '
+                    'return type must be specified.'
+                )
+
+        if not arg_types:
+            try:
+                arg_types = list(argspec.annotations[varname] for varname in argspec.args)
+            except KeyError:
+                raise ValueError(
+                    'Block callables must be fully annotated or explicit '
+                    'argument types must be specified.'
+                )
+        signature = tuple(ctype_for_type(tp) for tp in arg_types)
+
+        restype = ctype_for_type(restype)
+
+        self.cfunc_type = CFUNCTYPE(restype, c_void_p, *signature)
+
+        self.literal = BlockLiteral()
+        self.literal.isa = _NSConcreteGlobalBlock.ptr
+        self.literal.flags = BlockConsts.HAS_STRET | BlockConsts.HAS_SIGNATURE
+        self.literal.reserved = 0
+        self.cfunc = self.cfunc_type(self.wrapper)
+        self.literal.invoke = cast(self.cfunc, c_void_p)
+        self.descriptor = BlockDescriptor()
+        self.descriptor.reserved = 0
+        self.descriptor.size = sizeof(BlockLiteral)
+
+        encoding_for_ctype(restype)
```

Comment:
This discards the return value from the `encoding_for_ctype` call - is this intentional? If so, it would be good to add a comment.

Reply:
looks like leftover debugging code from me.